auction-server/src/opportunity/entities/opportunity_evm.rs

```diff
@@ -23,8 +34,44 @@ pub struct OpportunityEvm {
     pub target_call_value: U256,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct OpportunityCreateEvm {
+    pub core_fields: OpportunityCoreFieldsCreate<TokenAmountEvm>,
+
+    pub target_contract:   ethers::abi::Address,
+    pub target_calldata:   Bytes,
+    pub target_call_value: U256,
+}
+
 impl Opportunity for OpportunityEvm {
     type TokenAmount = TokenAmountEvm;
+    type ModelMetadata = repository::OpportunityMetadataEvm;
+    type OpportunityCreate = OpportunityCreateEvm;
+
+    fn new_with_current_time(val: Self::OpportunityCreate) -> Self {
```

Comment:
nit: we can say `fn new_with_current_time(val: OpportunityCreateEvm) -> Self` to avoid one level of indirection